core/procedures.js

```diff
@@ -358,9 +424,58 @@ Blockly.Procedures.editProcedureCallback_ = function(block) {
       return;
     }
     block = innerBlock;
+  } else if (!block.type == Blockly.PROCEDURES_PROTOTYPE_BLOCK_TYPE){
```

Comment:
The above logic checks whether the right inputs, connections, etc. exists.  It can go into `getPrototypeBlock`.  Then this becomes

```
if (block.type == Blockly.PROCEDURES_DEFINITION_BLOCK_TYPE ||
    block.type == Blockly.PROCEDURES_CALL_BLOCK_TYPE) {
  block = Blockly.Procedures.getPrototypeBlock(block.getProcCode(), block.workspace)
}
```

Reply:
Unfortunately the define block can't get the procCode, so we can't use getPrototypeBlock when all you have is the definition block.

Reply:
Good point.